backend/api/src/user/main.ts

```diff
@@ -0,0 +1,42 @@
+import createUser from "./createUser";
+import listAllUsers from "./listAllUsers";
+import listUsers from "./listUsers";
+import getUserById from "./getUserById";
+import updateUser from "./updateUser";
+import deleteUser from "./deleteUser";
+import getUserByName from "./getUserByName";
+
+type AppSyncEvent = {
+  info: {
+    fieldName: string;
+  };
+  arguments: {
+    user: User;
+    userId: string;
+    userName: string;
+    input: ListUsersInput;
+  };
+};
+
+export async function handler(
+  event: AppSyncEvent
+): Promise<User[] | UsersWithPaginationParams | User | string | null> {
+  switch (event.info.fieldName) {
```

Comment:
- The main.ts file would be removed and instead of call main.handler, users' function would be called directly:

Example for listAllUsers:

backend/api/lib/serviceStack.ts
`
dataSources: {
        users: "src/user/main.handler",
        listAllUsers: "src/user/listAllUsers.handler",
},
resolvers: {
        "Query    listAllUsers": "listAllUsers",
        "Query    listUsers": "users"
`

backend/api/src/user/listAllUsers.ts
`
export async function handler(): Promise<User[]> {
  const params = {
    TableName: process.env.USERS_TABLE as string,
  };

  try {
    const { Items } = await dynamoDb.scan(params).promise();
    return Items as User[];
  } catch (error) {
    throw new Error(sanitizeError(JSON.stringify(error)));
  }
}
`

- The main.ts file would be kept and we would have different functions to resolve each graphQL:

backend/api/lib/serviceStack.ts
`
dataSources: {
        users: "src/user/main.handler",
        listAllUsers: "src/user/main.handlerListAllUsers",
},
resolvers: {
        "Query    listAllUsers": "listAllUsers",
        "Query    listUsers": "users",
`

backend/api/src/user/main.ts
`
export async function handlerListAllUsers(): Promise<User[]> {
  return await listAllUsers();
}
`